src/v/redpanda/debug_bundle.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "rpk_errc.h"
+#include "seastarx.h"
+#include "utils/gate_guard.h"
+#include "utils/mutex.h"
+#include "utils/request_auth.h"
+
+#include <seastar/core/future.hh>
+#include <seastar/core/gate.hh>
+#include <seastar/core/iostream.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/core/sstring.hh>
+#include <seastar/util/process.hh>
+
+#include <concepts>
+#include <filesystem>
+#include <memory>
+#include <optional>
+#include <stdexcept>
+#include <system_error>
+#include <vector>
+
+static constexpr ss::shard_id debug_bundle_shard_id = 0;
+
+struct debug_bundle_credentials {
+    ss::sstring username;
+    ss::sstring password;
+    ss::sstring mechanism;
+};
+
+struct debug_bundle_params {
+    std::optional<ss::sstring> logs_since;
+    std::optional<ss::sstring> logs_until;
+    std::optional<ss::sstring> logs_size_limit;
+    std::optional<ss::sstring> metrics_interval;
```

Comment:
> should we pass non-string-based types here? the issue is that this interface creates coupling in that the caller (e.g. tests, admin_server) must understand how to format rpk arguments. 

A request with `logs_since` or `logs_until` uses journalctl format ("year-month-day"). So, if we force `logs_since` and `logs_until` to be a datetime object, then the admin_server will have to understand the journalctl format to convert to datetime. Therefore, I do not believe switching to non-string-based types changes the degree of coupling.

The same would be true for:
- `logs_size_limit` which takes the form "100MiB" or "1GB" for example, so the admin server must understand the memory units
- `metrics_interval` which takes the form "10s" or "1min" for example, so the admin server must convert that arg to the appropriate chrono type such as `std::chrono::seconds`.

Reply:
> the issue is that this interface creates coupling in that the caller (e.g. tests, admin_server) must understand how to format rpk arguments

I am also confused by this point. Tests and users will have to understand how to format rpk arguments regardless when submitting REST requests to the Admin API. The PRD explictly states the format which users must use to submit these values. 

In regards to the admin server, as it is now, the server simply forwards the args to RPK.

Reply:
> A request with logs_since or logs_until uses journalctl format ("year-month-day"). So, if we force logs_since and logs_until to be a datetime object, then the admin_server will have to understand the journalctl format to convert to datetime. Therefore, I do not believe switching to non-string-based types changes the degree of coupling.

here is an example of what i mean by reducing coupling

```
// typed interface
function rpk(std::chrono::system_clock::time_point since) {
   std::string since_str = convert_to_format_rpk_expects(since);
   ...
}

// admin server defines the format of a time for its json interface
function admin_server_handler(request) {
    auto since_str = request.get("since");
    auto since = convert_to_timepoint(since);
    // ^ provides validation at the edge
    rpk(since);
}

// tests can be written using normal c++ std::chrono types
function test() {
    auto since = std::chrono::system_clock::now() - std::chrono::hours(1);
    rpk(since);
}
```

Reply:
I see your point for the datetime arguments, however I really do not see how we could take a lower-level type for `log_size_limit` beyond removing the bytes type ("GB", "MB", etc) and then validating that the number is an integer. IMO, we can simply leave that as a string, no?

Reply:
> I see your point for the datetime arguments, however I really do not see how we could take a lower-level type for `log_size_limit` beyond removing the bytes type ("GB", "MB", etc) and then validating that the number is an integer. IMO, we can simply leave that as a string, no?

We could use a regex to check it, but I would check with `rpk` to see what kind of validation it does for this.

Reply:
it's a minor thing so of course we can leave it as strings. but consider that by changing it you'll make the system easier to test and understand. for example, how would I write a test like this?

```
    auto since = std::chrono::system_clock::now() - std::chrono::hours(1);
    rpk(since);
```

I think i'd have to:

1. go look at the implementation to see what kind of format is needed
2. figure out how to convert the timepoint into that format
3. fix all the tests if the format ever changed